upgrades/upgrade_testing_framework/steps/step_report_results.py

```diff
@@ -0,0 +1,115 @@
+                # failed, skipped, and passed, and it should greedily be applied to the most "serious" one,
+                # so failed, skipped, passed in that order. That means that if there are multiple assertions
+                # and they have different results (perhaps the first one passes, the second fails, and that
+                # causes the third to be skipped), it will show up as though the entire test failed.
+                if results['fail'] > 0:
+                    failed_expectations.append((expectation_id, test_phase))
+                    continue
+                if results['skip'] > 0:
+                    skipped_expectations.append((expectation_id, test_phase))
+                    continue
+                if results['pass'] > 0:
+                    passed_expectations.append((expectation_id, test_phase))
+                    continue
+        log_results(self.logger, passed_expectations, skipped_expectations, failed_expectations, expectations)
+        self.logger.info("For additional details on test results, see the test "
+                         f"result directory: {test_output_directory.absolute()}")
+
+        # Update our state
+        # N/A
+
+
+def parse_robot_results(test_output_dir: Path, expectation_ids: List[str]) -> Dict[str, Dict]:
+    """ This function takes a path to the Robot framework test results directory and a list of expectation_ids.
+    It expects a directory structure where there are subdirectories that correspond to different test phases
+    (e.g. pre-upgrade and post-upgrade). Each of these directories may have an `output.xml` file that contains
+    the results for each test.
+    """
+    results = {e: {} for e in expectation_ids}
+    for test_dir in test_output_dir.iterdir():
+        if not test_dir.is_dir():
+            continue
+        output_file = test_dir / "output.xml"
+        if not output_file.exists():
+            continue
+        test_phase_name = test_dir.parts[-1]
+
+        with output_file.open('r') as f:
+            tree = ET.parse(f)
+
+        root = tree.getroot()
+        stats_by_tag = {tag.text: {k: int(v) for k, v in tag.items()} for tag in root.find('statistics/tag')}
+
+        for expectation_id in expectation_ids:
+            if expectation_id in stats_by_tag:
+                results[expectation_id][test_phase_name] = stats_by_tag[expectation_id]
+    return results
+
+
+def log_result(logger, expectation_id, phases, expectation_description=None):
+    logger.info(expectation_id)
+    if None not in phases:
+        logger.info(f"\tPhases: {', '.join(phases)}")
+    if expectation_description:
+        for line in textwrap.wrap(expectation_description, width=120,
+                                  initial_indent="\tDescription: ", subsequent_indent="\t"):
+            logger.info(line)
+    logger.info("\n")
+
+
+def log_results(logger, passed: List[tuple], skipped: List[tuple], failed: List[tuple], expectations: Dict):
```

Comment:
I think we want to make an actual report of the results as well as logging them.  Open to the format, but maybe a JSON file to start off with and eventually make an HTML file?

Reply:
@chelma Is there anything you are looking to put in an eventual HTML file that doesn't exist in the existing robot framework output files? I think we could do some minor modifications to these files as well but not looked into how extensive these are

Reply:
As this grows as well (and depending what we list out on the actual robot framework test executions) we may only want to list failed and skipped expectations

Reply:
Yeah - really not sure what the robot tests give us out-of-the-box.  Will investigate.

Reply:
I found what I think is a good compromise with regards to STDOUT logging; will be interested to see what you think when I post my next revision.  The Robot Framework `log.html` and `report.html` files are probably sufficient on their own as long as top-level results are surfaced to STDOUT.